solutions/components-in-depth/src/app/timer/timer.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core'
+
+const MS_IN_SECOND = 1000;
+
+@Component({
+  selector: 'app-timer',
+  templateUrl: './timer.component.html',
+  styleUrls: ['./timer.component.css']
+})
+export class TimerComponent {
+  @Input() interval: number = 50;
+  @Output() onInterval = new EventEmitter();
+  @Output() onReset = new EventEmitter();
+  @Output() onStop = new EventEmitter();
+  private time: number = 0;
+  private timerId: NodeJS.Timer;
```

Comment:
First time seeing the `NodeJS.Timer` type but I wonder if we can simplify it just using `number` as the type?